src/Aspire.Hosting.Sdk/SDK/Sdk.targets

```diff
@@ -13,34 +15,92 @@
+  </Target>
+
+  <Target Name="__WarnOnMininumVsVersionMissing" BeforeTargets="PrepareForBuild"
+    Condition="'$(BuildingInsideVisualStudio)' == 'true' and $([MSBuild]::VersionLessThan('$(MSBuildVersion)', '17.10.0'))">
+    <Warning Code="ASPIRE003"
+      Text="$(MSBuildProjectName) is a .NET Aspire AppHost project that requires Visual Studio version 17.10 or above to work correctly. You are using version $(MSBuildVersion)." />
   </Target>
 
-  <Target Name="__WarnOnMininumVsVersionMissing" BeforeTargets="PrepareForBuild" Condition="'$(BuildingInsideVisualStudio)' == 'true' and $([MSBuild]::VersionLessThan('$(MSBuildVersion)', '17.10.0'))">
-    <Warning Code="ASPIRE003" Text="$(MSBuildProjectName) is a .NET Aspire AppHost project that requires Visual Studio version 17.10 or above to work correctly. You are using version $(MSBuildVersion)." />
+  <!-- This target extracts the version of Aspire.Hosting.AppHost referenced by the project, and adds
+       a reference to Aspire.Dashboard.Sdk and Aspire.Hosting.Orchestration for the build-time platform using
+       the same version. This is done here dynamically to avoid having to pull in DCP and Dashboard packages
+       for all of the platforms. This mechanism can be disabled by setting `SkipAddReferenceToDashboardAndDCP` to `true` -->
+  <Target Name="AddReferenceToDashboardAndDCP" BeforeTargets="Restore;CollectPackageReferences" Condition="'$(SkipAddReferenceToDashboardAndDCP)' != 'true'">
+
+    <!-- First, we assume project is not using Central Package Management, and we try to extract the version
+         from PackageReference Items. -->
+    <ItemGroup>
+      <AppHostPackageReference Include="@(PackageReference)" Condition="'%(Identity)' ==
+  'Aspire.Hosting.AppHost'" />
+    </ItemGroup>
+
+    <!-- Extract the Version metadata -->
+    <PropertyGroup>
+      <AppHostVersion>%(AppHostPackageReference.Version)</AppHostVersion>
+    </PropertyGroup>
+
+    <!-- If the version is still empty, then we assume the project is using Central Package Management, 
+         and we try to extract the PackageVersion Items. -->
+    <ItemGroup Condition="'$(AppHostVersion)' == ''">
+      <AppHostPackageReference />
+      <AppHostPackageReference Include="@(PackageVersion)" Condition="'%(Identity)' ==
+  'Aspire.Hosting.AppHost'" />
+    </ItemGroup>
+
+    <PropertyGroup Condition="'$(AppHostVersion)' == ''">
+      <AppHostVersion>%(AppHostPackageReference.Version)</AppHostVersion>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(AppHostVersion)' != ''">
+      <__CurrentAppHostVersionMessage> You are using version $(AppHostVersion).</__CurrentAppHostVersionMessage>
+    </PropertyGroup>
+
+    <!-- At this point, we should have the version either by CPM or PackageReference, so we fail if not.  -->
+    <Error Condition="'$(AppHostVersion)' == '' or $([MSBuild]::VersionLessThan('$(AppHostVersion)', '8.2.0'))" 
```

Comment:
nit: It might be useful to split this into two separate error messages, to help with debugging when this fails.

1. `AppHostVersion==''`, we could not figure out the version to use, and have a better error message in case of CPM.
2. `AppHostVersion < 8.2.0`

Reply:
We would know at that point. If 1., then the warning will not have the version you are using (see line 90 above). If 2 then you'd see which version you are using in the error message.